packages/headless/src/index.ts

```diff
@@ -207,7 +208,10 @@ export default async (opts: HeadlessOptions) => {
     registries: opts.registries,
     skipped,
   }
-  const filteredLockfile = filterLockfileByImportersAndEngine(wantedLockfile, opts.projects.map(({ id }) => id), {
```

Comment:
if this is the only place where `opts.projects` is used then probably just pass in the projectIds instead. Then no need in allImporterIds

Reply:
When `package.json` was presented, it seems like `opt.projects` will be empty array.